### DISCUSSION TITLE
Any way to prevent LSP from trying to attach in octo buffers?

### DISCUSSION THREAD

**Discussion by joshpetit** [pwntester/octo.nvim#225] [Q&A]
Octo is awesome! One thing that's a bit annoying though are the consistent errors from my lsp trying to attach to the octo buffers when reviewing. I consistently get this error:

```
LSP[dartls] URI was not a valid file:// URI
```

Is there some way to make the lsp not attach to the octo buffers?

**Comment by pwntester** [comment]
I use something like: https://github.com/pwntester/dotfiles/blob/master/config/nvim/lua/pwntester/lsp.lua#L139-L144 to prevent LSP from attaching to special buffers

**Reply by joshpetit** [reply]
Hm, does this work for every language server? It seems to not through exceptions for the dartls, but for tsserver it does. My dartls still attaches though :sweat_smile:. I could look into this more later but it doesn't stop me from using octo which makes it ok.

**Comment by Davincible** [comment]
@pwntester on the contrary, is LSP supposed to attach by default? It doesn't for me but I would like to have my LSP when doing reviews

**Reply by joshpetit** [reply]
If your LSP were to attach when doing reviews it would show a bunch of errors because the diff files aren't located in the same place as your project files.

**Reply by delabere** [reply]
I would really like to be able to use my LSP when looking at PR reviews after checking out the review branch locally. I can see in the [top post on this reddit thread](https://www.reddit.com/r/neovim/comments/ocgv0o/octonvim_github_issuesprs_from_neovim/) that OP (@pwntester) says it's possible to do so?

Any way this is possible?

**Reply by NWVi** [reply]
@delabere @Davincible If you want to be able to use your LSP when looking at PR reviews you can uncomment the `opts.use_local` check found here:
https://github.com/pwntester/octo.nvim/blob/b75630f93822a569f3fc2360bb2066ec1b205bd0/lua/octo/reviews/file-entry.lua#L394

I believe it was changed to support commit based review per the commit that removed it: https://github.com/pwntester/octo.nvim/commit/c3508b5f82c909ffec37b6dcac4227ea672161f0

I created a fork where I made this change if you want to use it. 
https://github.com/NWVi/octo.nvim

**Reply by pwntester** [reply]
Yep, exactly, I think it would be good to make a config option so you could use the contents from the file system (will require changing branches and that change may not be reverted depending on how vim is exited) or use it from git store. @NWVi, would you be able to send a PR for this?

**Reply by NWVi** [reply]
@pwntester I made an attempt at a PR for this. My implementation might be a little naive, and does not revert changing branches after review or exiting vim if the user accepts the PR branch checkout.

https://github.com/pwntester/octo.nvim/pull/349

**Reply by pwntester** [reply]
Awesome! thanks!
